lib/presentation/screens/images_screen/images_screen_layout.dart

```diff
@@ -1,38 +1,41 @@
-import 'package:array_names/business_logic/images_provider.dart';
 import 'package:array_names/data/models/breed.dart';
+import 'package:array_names/presentation/widgets/images_widget.dart';
 import 'package:flutter/material.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
 
-class ImagesLayout extends StatelessWidget {
+import '../../../business_logic/images_provider.dart';
+
+class ImagesLayout extends ConsumerStatefulWidget {
   final Breed breed;
+
   const ImagesLayout({Key? key, required this.breed}) : super(key: key);
 
+  @override
+  ConsumerState<ConsumerStatefulWidget> createState() => ImagesLayoutState();
+}
+
+class ImagesLayoutState extends ConsumerState<ConsumerStatefulWidget> {
+  @override
+  void initState() {
+    ref.read(getImagesProvider.notifier).getImages(
+          (widget as ImagesLayout).breed,
```

Comment:
just a suggestion, but you could make field called parentWidget or something, assign here `widget as ImagesLayout` and there will be no need in cast every time